frontend/src/app/patients/personSchema.ts

```diff
@@ -0,0 +1,83 @@
+import * as yup from "yup";
+import { PhoneNumberUtil } from "google-libphonenumber";
+
+const phoneUtil = PhoneNumberUtil.getInstance();
+
+type PartialBy<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>;
+
+type OptionalFields =
+  | "middleName"
+  | "lookupId"
+  | "role"
+  | "email"
+  | "streetTwo"
+  | "city"
+  | "county"
+  | "race"
+  | "ethnicity"
+  | "gender"
+  | "residentCongregateSetting"
+  | "employedInHealthcare";
+
+export type RequiredPersonFields = PartialBy<
+  Nullable<PersonFormData>,
+  OptionalFields
+>;
+
+export const personSchema: yup.SchemaOf<RequiredPersonFields> = yup.object({
+  firstName: yup.string().required(),
+  middleName: yup.string().nullable(),
+  lastName: yup.string().required(),
+  lookupId: yup.string().nullable(),
+  role: yup.string().nullable() as any,
```

Comment:
I see a few `string()` fields here `as any` but I think `oneOf` would work:

```
role: yup
  .mixed()
  .oneOf(["STAFF", "RESIDENT", "STUDENT", "VISITOR", ""])
  .nullable(),
```

Reply:
Didn't know this was possible! this is much better.

Turns out that `nullable` doesn't work and the null needs to be enumerated in the array passed to `oneOf`